src/reducers/index.js

```diff
@@ -511,17 +522,25 @@ export const resetSelectedTokenIfNeeded = (state, action) => {
   return state;
 };
 
-/*
+/**
  * Used when registering or creating tokens to update the wallet token list.
-*/
+ * @param {Record<string,string>} state.allTokens - Current list of allTokens
+ * @param {string} action.payload.uid - UID of token to add
+ */
 export const onNewTokens = (state, action) => {
+  // Convert `allTokens` to a Set to prevent duplicates
+  const allTokensSet = new Set();
+  for (const token of Object.keys(state.allTokens)) {
+    allTokensSet.add(token);
+  }
   // Add new created token to the all tokens set
-  state.allTokens.add(action.payload.uid);
+  allTokensSet.add(action.payload.uid);
```

Comment:
I think this is better:
```js
  const allTokensSet = new Set(Object.keys(state.allTokens));
```

But if this is just to convert to set and remove possible duplicates maybe we could just do:

```js
state.allTokens[action.payload.uid] = action.payload.uid;
```

This way we don't need to convert to set and have the same effect.

There is also:

```js
 const newAllTokens = {...state.allTokens, [action.payload.uid]: action.payload.uid};
```

Now we create a new object, so redux can detect the change.

Reply:
My implementation on https://github.com/HathorNetwork/hathor-wallet/pull/587/commits/3ee5ac5cfba3a1e7dbe5387109159d06d4a34485 was close to this one, but with more steps for clarity.

What do you think?

Reply:
I usually prefer oneliners, but splitting for better readability is always nice!